FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_IP.c

```diff
@@ -2289,7 +2300,7 @@ size_t uxDataLengthBytes = uxByteCount;
 	/* coverity[value_overwrite] */
 	xSum.u32 = ( uint32_t ) xSum.u16[ 0 ] + xSum.u16[ 1 ];
 
-	if( ( ulAlignBits & 1U ) != 0U )
+	if( ( uxAlignBits & 1U ) != 0U )
```

Comment:
The logic on lines 2304-2308 appears identical to the new logic added in lines 2210-2212.  Is this block still required?

Reply:
My understanding of this part of the code is that it swaps the endianness of the lower half of the accumulator back to that of the machine.
The order of the bytes in the accumulator depends on both the endianness of the machine and the alignment of the packet. So if the packet is aligned on an odd boundary then the bytes in the two half words of the accumulator are in different orders to each other.
This seems to be an interaction between Parts (B) and (C) of https://tools.ietf.org/html/rfc1071#page-3